core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -224,54 +241,75 @@ private[spark] class ExecutorAllocationManager(
   }
 
   /**
+   * Check to see whether our existing allocation and the requests we've made previously exceed our
+   * current needs. If so, let the cluster manager know so that it can cancel pending requests that
+   * are unneeded.
+   *
+   * If not, and the add time has expired, see if we can request new executors and refresh the add
+   * time.
+   *
+   * @return the delta in the target number of executors.
+   */
+  private def adjustRequestedExecutors(now: Long): Int = synchronized {
+    val currentTarget = targetNumExecutors
+    val maxNeeded = maxNumExecutorsNeeded
+
+    if (maxNeeded < currentTarget) {
+      // The target number exceeds the number we actually need, so stop adding new
+      // executors and inform the cluster manager to cancel the extra pending requests.
+      client.requestTotalExecutors(maxNeeded)
+      val delta = math.min(maxNeeded - currentTarget, 0)
```

Comment:
why are you doing `math.min` here? Since `maxNeeded < currentTarget`, `maxNeeded - currentTarget` is always going to be negative.